airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -166,19 +267,19 @@ def _get_filetype(self, file: IOBase, remote_file: RemoteFile) -> Optional[FileT
     def _supported_file_types(self) -> List[Any]:
         return [FileType.MD, FileType.PDF, FileType.DOCX, FileType.PPTX]
 
-    def _render_markdown(self, elements: List[Any]) -> str:
+    def _render_markdown(self, elements: List[Dict[str, Any]]) -> str:
         return "\n\n".join((self._convert_to_markdown(el) for el in elements))
 
-    def _convert_to_markdown(self, el: Any) -> str:
-        if isinstance(el, Title):
-            heading_str = "#" * (el.metadata.category_depth or 1)
-            return f"{heading_str} {el.text}"
-        elif isinstance(el, ListItem):
-            return f"- {el.text}"
-        elif isinstance(el, Formula):
-            return f"```\n{el.text}\n```"
+    def _convert_to_markdown(self, el: Dict[str, Any]) -> str:
+        if dpath.util.get(el, "type") == "Title":
+            heading_str = "#" * (dpath.util.get(el, "metadata/category_depth", default=1) or 1)
+            return f"{heading_str} {dpath.util.get(el, 'text')}"
+        elif dpath.util.get(el, "type") == "ListItem":
+            return f"- {dpath.util.get(el, 'text')}"
+        elif dpath.util.get(el, "type") == "Formula":
+            return f"```\n{dpath.util.get(el, 'text')}\n```"
         else:
-            return str(el.text) if hasattr(el, "text") else ""
+            return str(dpath.util.get(el, "text", default=""))
```

Comment:
TIL `dpath` is a thing 😄 

https://pypi.org/project/dpath/

For my info, what was the driver for moving to dpath for this method? (Just curious.)